### PR TITLE
Fix working on constraints, labelontop, editable when list is sorted

### PR DESCRIPTION
**Pull request by tudorbarascu**
Hello,

If the field list is sorted by name, the ``setLabelOnTop``, ``*FieldConstraint``, ``setReadOnly`` functions should take into consideration the sorted list IDs instead of the unsorted one.

This also needs backporting.
There's still an issue that I noticed while fixing this:

The changes made by the ``setReadOnly`` and ``setLabelOnTop`` functions work well after these commits but the changes don't get saved in the project. Any hints towards solving this?

Thanks a lot!
Tudor

### CODE REVIEW

File: src/app/qgsfieldsproperties.cpp [hunk 1]
```diff
@@ -989,34 +989,34 @@ void QgsFieldsProperties::apply()
     QString name = mLayer->fields().at( idx ).name();
     FieldConfig cfg = configForRow( i );
 
-    editFormConfig.setReadOnly( i, !cfg.mEditable );
-    editFormConfig.setLabelOnTop( i, cfg.mLabelOnTop );
+    editFormConfig.setReadOnly( idx, !cfg.mEditable );
+    editFormConfig.setLabelOnTop( idx, cfg.mLabelOnTop );
     mLayer->setConstraintExpression( i, cfg.mConstraint, cfg.mConstraintDescription );
```
Comment (m-kuhn) [inline src/app/qgsfieldsproperties.cpp]: Are constraints also affected?

Comment (tudorbarascu) [inline src/app/qgsfieldsproperties.cpp]: no, constraints are solved by the commit; they get saved ok

Comment (m-kuhn) [inline src/app/qgsfieldsproperties.cpp]: And on this line (994), shouldn't `i` be replaced with `idx`?

Comment (tudorbarascu) [inline src/app/qgsfieldsproperties.cpp]: yes it should. Somehow I missed it. thanks for your keen eye Matthias

**Pull request merged**